src/Security/LoginFormAuthenticator.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace App\Security;
+
+use App\Entity\User;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Encoder\UserPasswordEncoderInterface;
+use Symfony\Component\Security\Core\Exception\InvalidCsrfTokenException;
+use Symfony\Component\Security\Core\Security;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Symfony\Component\Security\Core\User\UserProviderInterface;
+use Symfony\Component\Security\Csrf\CsrfToken;
+use Symfony\Component\Security\Csrf\CsrfTokenManagerInterface;
+use Symfony\Component\Security\Guard\Authenticator\AbstractFormLoginAuthenticator;
+use Symfony\Component\Security\Http\Util\TargetPathTrait;
+
+class LoginFormAuthenticator extends AbstractFormLoginAuthenticator
```

Comment:
I am not against promoting the recommended form, but this class has many complex steps and concepts compared to the previous code. Unless you need custom behavior in the login process, I think this class is very difficult for newcomers.

Reply:
You are right that this is more complex than before ... but if we consider that this is "too complex" ... then the problem lies in Symfony, which proposes a solution that is too complex.

@weaverryan what's your opinion? Should we leave this login form authenticator outside of this demo app? Thanks!

Reply:
Yep, this feels like a lot of *manual* statements (and a lot of knowledge) that are automatic with previous config, IMHO.
> Unless you need custom behavior in the login process